src/porepy/numerics/fv/biot.py

```diff
@@ -1079,14 +1103,14 @@ def assemble_rhs(self, g, data):
         # transient BCs, use the below with the appropriate version of d_bound_i.
         d_bound_1 = parameter_dictionary["bc_values"]
 
-        d_bound_0 = parameter_dictionary["state"]["bc_values"]
+        d_bound_0 = data[pp.STATE][self.keyword]["bc_values"]
         biot_alpha = parameter_dictionary["biot_alpha"]
         rhs_bound = (
             -matrix_dictionary["bound_div_d"] * (d_bound_1 - d_bound_0) * biot_alpha
         )
 
         # Time part
-        d_cell = parameter_dictionary["state"]["displacement"]
+        d_cell = data[pp.STATE][self.variable]
```

Comment:
You have data[pp.STATE][self.keyword]["bc_values"], but here it is only data[pp.STATE][self.variable]. I understand why, but it is a bit inconsistent